### PR TITLE
feat: add cri

### PR DESCRIPTION
**Pull request by PrashantRaj18198**

### CODE REVIEW

File: pkg/app/app.go [hunk 1]
```diff
@@ -101,6 +111,8 @@ func NewApp(ctx context.Context, opts *ApplicationOptions) error {
 			settingsData, err := os.ReadFile(opts.SettingsFile)
 			if err != nil {
 				if os.IsNotExist(err) {
+					// TODO: remove this
```
Comment (prajjwaldimri) [inline pkg/app/app.go:114]: Unresolved TODO

File: pkg/manager/agent.go [hunk 2]
```diff
@@ -111,6 +111,15 @@ func (a *agentImpl) StartAgent(ctx context.Context, opts *StartAgentOptions) err
 					os.Setenv(key, val)
 				}
 
+				if opts.Manager.Provider == ProviderGithubCRI {
+					for key, val := range *allocationDetails.GhRunnerApplicationDetails.Variables {
```
Comment (prajjwaldimri) [inline pkg/manager/agent.go:115]: Add null checks for this dereference if not present

File: pkg/manager/manager_github_cri.go [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+func (m *ghcriManager) StartRunner(ctx context.Context, opts *StartRunnerOptions) (*StartRunnerOutput, error) {
+	err := m.init(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// containerID := opts.AgentOptions.ID
+
+	// TODO: verify if the pull is needed or not for the image.
```
Comment (prajjwaldimri) [inline pkg/manager/manager_github_cri.go:64]: Resolve TODOs and remove comments

File: pkg/manager/manager_github_cri.go [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+	for {
+		select {
+		case out := <-stdoutChan:
+			fmt.Fprintln(stdoutFile, out)
+			fmt.Fprintln(os.Stdout, out)
+		case err := <-stderrChan:
+			fmt.Fprintln(stderrFile, err)
+			fmt.Fprintln(os.Stderr, err)
+		case <-ticker.C:
+			// Handle output every second
```
Comment (prajjwaldimri) [inline pkg/manager/manager_github_cri.go:197]: Nothing is happening on this ticker. Should we remove this?

**Review by prajjwaldimri — CHANGES_REQUESTED** [review]

File: pkg/manager/agent.go (hunk 2 above)
Comment (PrashantRaj18198) [reply pkg/manager/agent.go:120]: Added above this.

File: pkg/app/app.go (hunk 1 above)
Comment (PrashantRaj18198) [reply pkg/app/app.go]: resolved.

File: pkg/manager/manager_github_cri.go (hunk 4 above)
Comment (PrashantRaj18198) [reply pkg/manager/manager_github_cri.go]: removed

File: pkg/manager/manager_github_cri.go (hunk 3 above)
Comment (PrashantRaj18198) [reply pkg/manager/manager_github_cri.go]: removed

**Review by prajjwaldimri — APPROVED** [review]
LGTM 🪀

Please remove that ticker code from `manager_github` also if possible

**Comment by PrashantRaj18198** [comment]
> Please remove that ticker code from manager_github also if possible

Can't remove that just yet. We'll take this up later.

**Pull request merged**